src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Debugger_InProc.cs

```diff
@@ -33,5 +56,105 @@ public void SetBreakPoint(string fileName, int lineNumber, int columnIndex)
+
+        /// <summary>
+        /// Executes the specified action delegate and retries if Operation Not Supported is thrown.
+        /// </summary>
+        /// <param name="action">Action delegate to exectute.</param>
+        private void WaitForRaiseDebuggerDteCommand(Action action)
+        {
+            bool actionSucceeded = false;
+
+            Func<bool> predict = delegate
+            {
+                try
+                {
+                    action();
+                    actionSucceeded = true;
+                }
+                catch (COMException ex)
+                {
+                    if ((uint)ex.ErrorCode != OperationNotSupportedHResult)
+                    {
+                        string message = string.Format(
+                            CultureInfo.InvariantCulture,
+                            "Failed to raise debugger command, an unexpected '{0}' was thrown with the HResult of '{1}'.",
+                            typeof(COMException),
+                            ex.ErrorCode);
+
+                        throw new Exception(message, ex);
+                    }
+
+                    actionSucceeded = false;
+                }
+
+                return actionSucceeded;
+            };
+
+            // Repeat the the command if "Operation Not Supported" is thrown.
+            if (!this.TryWaitFor(DebuggerCommandRetryTimeout, predict))
+            {
+                string message = string.Format(
+                    CultureInfo.InvariantCulture,
+                    "Failed to raise debugger command within '{0}' seconds.",
+                    DebuggerCommandRetryTimeout.TotalSeconds);
+
+                throw new Exception(message);
+            }
+        }
+
+        /// <summary>
+        /// Polls for the specified delegate to return true for the given timeout.
+        /// </summary>
+        /// <param name="timeout">Timeout to keep polling.</param>
+        /// <param name="predicate">Delegate to invoke.</param>
+        /// <returns>True if the delegate returned true when polled, otherwise false.</returns>
+        public bool TryWaitFor(TimeSpan timeout, Func<bool> predicate)
+        {
+            return this.TryWaitFor(timeout, DefaultPollingInterCallSleep, predicate);
+        }
+
+        /// <summary>
+        /// Polls for the specified delegate to return true for the given timeout.
+        /// </summary>
+        /// <param name="timeout">Timeout to keep polling.</param>
+        /// <param name="interval">Time to wait between polling.</param>
+        /// <param name="predicate">Delegate to invoke.</param>
+        /// <returns>
+        /// True if the delegate returned true when polled, otherwise false.
+        /// </returns>
+        private bool TryWaitFor(TimeSpan timeout, TimeSpan interval, Func<bool> predicate)
+        {
+            DateTime endTime = DateTime.UtcNow + this.ApplyTimeoutMultiplier(timeout);
```

Comment:
>ApplyTimeoutMultiplier [](start = 54, length = 22)

Do we need this? The constant is 1.0f, so it doens't do anything.

Reply:
Sure. Let us remote it. Thanks!

---
In reply to: [136209556](https://github.com/dotnet/roslyn/pull/21824#discussion_r136209556) [](ancestors = 136209556)